ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/WritePackagesService.scala

```diff
@@ -21,18 +24,24 @@ trait WritePackagesService {
     * message. See the comments on [[ReadService.stateUpdates]] and [[Update]] for
     * further details.
     *
-    * @param payload           : DAML-LF packages to be uploaded to the ledger.
-    * @param sourceDescription : the description of the packages provided by the
+    * Note: we accept [[Archive]]s rather than parsed packages, because we want
+    * to be able to get the byte size of each individual ArchivePayload, which
+    * is information that the read / index service need to provide. Moreover
+    * this information should be consistent with the payload that the
+    * [[com.digitalasset.ledger.api.v1.package_service.GetPackageResponse]]
+    * contains. If we were to consume packages we'd have to re-encode them to
+    * provide the size, and the size might potentially be different from the
+    * original size, which would be quite confusing.
+    *
+    * @param sourceDescription : The description of the packages provided by the
     *                            participant implementation.
     *
-    * @return an async result of a SubmissionResult
+    * @param payload           : DAML-LF packages to be uploaded to the ledger.
```

Comment:
Since a package is something specific inside an `ArchivePayload`, should we phrase it as 
```
DAML-LF archives to be uploaded to the ledger
```
?